src/nvim/ops.c

```diff
@@ -2927,13 +2926,14 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
      */
     if (y_type == MCHAR && y_size == 1) {
       do {
-        totlen = count * yanklen;
+        assert(count <= INT_MAX);
```

Comment:
actually I tried to blame the `count == -1` line in vim but got stuck at 7.0.001 and I was to lazy to look up the archival repository. I suppose this was once used somewhere but that got deleted. Let's say assuming count >= 0 is fine.